### PR TITLE
[dagster-fivetran] Migrate to Pythonic resources

### PR DESCRIPTION
**Pull request by benpankow**
## Summary

Converts the `dagster-fivetran` internals to use the new config and resource systems:

- Assets and ops now use Pythonic config rather than config_schema
- Assets and ops retrieve resource dependencies through params rather than the `context` / `required_resource_keys`
- `FivetranResource` is now a `ConfigurableResource`, holding its own config schema. The `fivetran_resource` function resource now wraps this class rather than holding the config.

## Test Plan

Existing unit tests.

### CODE REVIEW

**Comment by benpankow** [comment]
Current dependencies on/for this PR:
* master
  * **PR #12975** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12975" target="_blank">Graphite</a> 
    * **PR #12976** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12976" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/12976?utm_source=stack-comment).

File: python_modules/libraries/dagster-fivetran/dagster_fivetran/ops.py [hunk 1]
```diff
@@ -1,12 +1,57 @@
...
+from dagster import (
+    AssetKey,
+    Config,
+    In,
+    Nothing,
+    Out,
+    Output,
+    op,
+)
+from pydantic import Field as PyField
```
Comment (schrockn) [inline python_modules/libraries/dagster-fivetran/dagster_fivetran/ops.py:12]: alias

File: python_modules/libraries/dagster-fivetran/dagster_fivetran/ops.py [hunk 2]
```diff
@@ -1,12 +1,57 @@
...
 
-from dagster_fivetran.resources import DEFAULT_POLL_INTERVAL
+from dagster_fivetran.resources import DEFAULT_POLL_INTERVAL, FivetranResource
 from dagster_fivetran.types import FivetranOutput
 from dagster_fivetran.utils import generate_materializations
 
 
+class SyncConfig(Config):
+    connector_id: str = PyField(
+        ...,
```
Comment (schrockn) [inline python_modules/libraries/dagster-fivetran/dagster_fivetran/ops.py:21]: elim unnecessary ellipsis

File: python_modules/libraries/dagster-fivetran/dagster_fivetran/ops.py [hunk 3]
```diff
@@ -207,24 +178,24 @@ def my_composed_fivetran_job():
...
-    fivetran_output = context.resources.fivetran.resync_and_poll(
-        connector_id=context.op_config["connector_id"],
-        resync_parameters=context.op_config["resync_parameters"],
-        poll_interval=context.op_config["poll_interval"],
-        poll_timeout=context.op_config["poll_timeout"],
+    fivetran_output = fivetran.resync_and_poll(
+        connector_id=config.connector_id,
+        resync_parameters=config.resync_parameters,
+        poll_interval=config.poll_interval,
+        poll_timeout=config.poll_timeout,
```
Comment (schrockn) [inline python_modules/libraries/dagster-fivetran/dagster_fivetran/ops.py:185]: this is most pleasing

**Review by schrockn — CHANGES_REQUESTED** [review]
pretty great showcase diff demonstrating the improve ergonomic

File: python_modules/libraries/dagster-fivetran/dagster_fivetran/resources.py [hunk 4]
```diff
@@ -448,11 +422,4 @@ def my_fivetran_job():
...
     """
-    return FivetranResource(
-        api_key=context.resource_config["api_key"],
-        api_secret=context.resource_config["api_secret"],
-        disable_schedule_on_trigger=context.resource_config["disable_schedule_on_trigger"],
-        request_max_retries=context.resource_config["request_max_retries"],
-        request_retry_delay=context.resource_config["request_retry_delay"],
-        log=context.log,
-    )
+    return FivetranResource(**context.resource_config).resource_fn(context)  # type: ignore
```
Comment (schrockn) [inline python_modules/libraries/dagster-fivetran/dagster_fivetran/resources.py:425]: use `from_resource_context`

**Review by schrockn — APPROVED** [review]
Use the new helper, but looks great otherwise 👍🏻

**Pull request merged**